# Here we have a data set showing the average temperature in cities across to clob from 1995 to 2020. The purpose of this project is to determine which states have had the highest increase in Average temperature in the United States.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('/Users/nickseah/Desktop/ClimateDataScienceProject/city_temperature.csv')
df.head()

/Users/nickseah/Desktop/PythonAddOns/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3071: DtypeWarning: Columns (2) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Region  Country State     City  Month  Day  Year  AvgTemperature
0  Africa  Algeria   NaN  Algiers      1    1  1995            64.2
1  Africa  Algeria   NaN  Algiers      1    2  1995            49.4
2  Africa  Algeria   NaN  Algiers      1    3  1995            48.8
3  Africa  Algeria   NaN  Algiers      1    4  1995            46.4
4  Africa  Algeria   NaN  Algiers      1    5  1995            47.9

# Create a USA dataframe that only contains cities in the United States of America

In [6]:
USA_df = df[df.Country == 'US']
USA_df.reset_index(drop = True, inplace=True)
USA_df.head()

Region Country    State        City  Month  Day  Year  \
0  North America      US  Alabama  Birmingham      1    1  1995   
1  North America      US  Alabama  Birmingham      1    2  1995   
2  North America      US  Alabama  Birmingham      1    3  1995   
3  North America      US  Alabama  Birmingham      1    4  1995   
4  North America      US  Alabama  Birmingham      1    5  1995   

   AvgTemperature  
0            50.7  
1            37.2  
2            33.2  
3            33.3  
4            26.4

# We will create 2 more dataframes. One for entries from 1995 and one from 2019. We aren't using 2020 because the year was not over when this data was sampled. From these two dataframes we will calculate the average change in temperare for each state.

In [38]:
USA_1995 = USA_df[USA_df['Year'] == 1995]
USA_2019 = USA_df[USA_df['Year'] == 2019]

# Aggregate by State column and calculate the average temperature for each state from 1995 

In [47]:
states_95 = USA_1995.groupby('State')
states_95 = states_95.agg(np.mean)
#states_95 = states_95.sort_values(by = 'AvgTemperature')
states_95 = states_95[1:]
states_95.head()

Month        Day    Year  AvgTemperature
State                                                  
Alabama     6.526027  15.720548  1995.0       63.436164
Alaska      6.526027  15.720548  1995.0       35.849772
Arizona     6.526027  15.720548  1995.0       66.849452
Arkansas    6.526027  15.720548  1995.0       61.439452
California  6.526027  15.720548  1995.0       61.239014

# Aggregate by State column and calculate the average temperature for each state from 2019

In [48]:
states_19 = USA_2019.groupby('State')
states_19 = states_19.agg(np.mean)
#states_19 = states_19.sort_values(by = 'AvgTemperature')
states_19.head()

Month        Day    Year  AvgTemperature
State                                                  
Alabama     6.526027  15.720548  2019.0       66.316849
Alaska      6.526027  15.720548  2019.0       40.147215
Arizona     6.526027  15.720548  2019.0       72.573973
Arkansas    6.526027  15.720548  2019.0       61.951644
California  6.526027  15.720548  2019.0       62.626082

# We will combine the 1995 and 2019 dataframes for easier analysis

In [57]:
combined_df = pd.merge(left = states_95, right = states_19, on='State',suffixes = ('95','19'))
combined_df['24YrTempIncrease'] = combined_df['AvgTemperature19']- combined_df['AvgTemperature95']
combined_df['24YrTempIncrease'].describe()

count    50.000000
mean      0.832469
std       2.195700
min      -8.589863
25%      -0.350616
50%       0.539589
75%       2.126908
max       5.724521
Name: 24YrTempIncrease, dtype: float64

# Using the describe function above we can see that the average state in the United states has increased by 0.83 degrees farenheit since 1995.